synapseclient/models/protocols/access_control_protocol.py

```diff
@@ -0,0 +1,104 @@
+"""Protocol for the specific methods of this class that have synchronous counterparts
+generatred at runtime."""
+
+from typing import Dict, List, Optional, TYPE_CHECKING, Union, Protocol
+from synapseclient import Synapse
+
+if TYPE_CHECKING:
+    from synapseclient.core.models.permission import Permissions
+
+
+class AccessControllableSynchronousProtocol(Protocol):
+    """
+    The protocol for methods that are asynchronous but also
+    have a synchronous counterpart that may also be called.
+    """
+
+    def get_permissions(
+        self,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "Permissions":
```

Comment:
why is this not the Self type hint

Reply:
Permissions is the correct type hint in this case. This is because [this is a mixin](https://github.com/Sage-Bionetworks/synapsePythonClient/blob/a079191c7e515f741472c69f82017991d50f80f7/synapseclient/models/mixins/access_control.py) that is used in other classes (Like Project for example) where the `Self` type hint would mean `Project` instead of `Permissions`.